### ISSUE TITLE
Kafka oauth authentication issue with  kafka-oauth-client 0.8.x versions 

### ISSUE THREAD

**Issue by zbendhiba** [quarkusio/quarkus#18398]
## Describe the bug
Hello, I'm using the kafka-client with Oauth, I've had an issue with [Athenticating with oAuth](https://quarkus.io/guides/kafka#authenticating-with-oauth)
Using the kafka-oauth-client 0.7.2 version --> it works fine
Using the latest kafka-oauth-client 0.8.x versions --> the dependency com.nimbusds:nimbus-jose-jwt is needed 
Stack trace :

```
2021-07-02 15:34:25,931 ERROR [io.qua.run.Application] (Quarkus Main Thread) Failed to start application (with profile dev): java.lang.NoSuchMethodError: 'java.util.Map com.nimbusds.jose.Payload.toJSONObject()'
        at io.strimzi.kafka.oauth.common.NimbusPayloadTransformer.transform(NimbusPayloadTransformer.java:15)
        at io.strimzi.kafka.oauth.common.NimbusPayloadTransformer.transform(NimbusPayloadTransformer.java:11)
        at com.nimbusds.jose.Payload.toType(Payload.java:495)
        at io.strimzi.kafka.oauth.common.TokenIntrospection.introspectAccessToken(TokenIntrospection.java:31)
        at io.strimzi.kafka.oauth.common.OAuthAuthenticator.post(OAuthAuthenticator.java:141)
        at io.strimzi.kafka.oauth.common.OAuthAuthenticator.loginWithClientSecret(OAuthAuthenticator.java:72)
        at io.strimzi.kafka.oauth.client.JaasClientOauthLoginCallbackHandler.handleCallback(JaasClientOauthLoginCallbackHandler.java:161)
        at io.strimzi.kafka.oauth.client.JaasClientOauthLoginCallbackHandler.handle(JaasClientOauthLoginCallbackHandler.java:141)
        at org.apache.kafka.common.security.oauthbearer.OAuthBearerLoginModule.identifyToken(OAuthBearerLoginModule.java:316)
        at org.apache.kafka.common.security.oauthbearer.OAuthBearerLoginModule.login(OAuthBearerLoginModule.java:301)
        at java.base/javax.security.auth.login.LoginContext.invoke(LoginContext.java:726)
        at java.base/javax.security.auth.login.LoginContext$4.run(LoginContext.java:665)
        at java.base/javax.security.auth.login.LoginContext$4.run(LoginContext.java:663)
        at java.base/java.security.AccessController.doPrivileged(Native Method)
        at java.base/javax.security.auth.login.LoginContext.invokePriv(LoginContext.java:663)
        at java.base/javax.security.auth.login.LoginContext.login(LoginContext.java:574)
        at org.apache.kafka.common.security.oauthbearer.internals.expiring.ExpiringCredentialRefreshingLogin.login(ExpiringCredentialRefreshingLogin.java:204)
        at org.apache.kafka.common.security.oauthbearer.internals.OAuthBearerRefreshingLogin.login(OAuthBearerRefreshingLogin.java:150)
        at org.apache.kafka.common.security.authenticator.LoginManager.<init>(LoginManager.java:62)
        at org.apache.kafka.common.security.authenticator.LoginManager.acquireLoginManager(LoginManager.java:105)
        at org.apache.kafka.common.network.SaslChannelBuilder.configure(SaslChannelBuilder.java:158)
        at org.apache.kafka.common.network.ChannelBuilders.create(ChannelBuilders.java:157)
        at org.apache.kafka.common.network.ChannelBuilders.clientChannelBuilder(ChannelBuilders.java:73)
        at org.apache.kafka.clients.ClientUtils.createChannelBuilder(ClientUtils.java:105)
        at org.apache.kafka.clients.producer.KafkaProducer.newSender(KafkaProducer.java:449)
        at org.apache.kafka.clients.producer.KafkaProducer.<init>(KafkaProducer.java:430)
        at org.apache.kafka.clients.producer.KafkaProducer.<init>(KafkaProducer.java:302)
```

**Comment by cescoffier** [comment]
can you provide a quick reproducer? 
I would like to see who's pulling the other nimbus version.

**Comment by zbendhiba** [comment]
the issue is related to camel-quarkus

**Comment by mstruk** [comment]
This issue happens with Quarkus native due to nimbus-jose-jwt dependency version mismatch.

Typically you control a consistent set of dependency versions for Quarkus application by importing a quarkus-platform-bom:

```
  <dependencyManagement>
    <dependencies>
      <dependency>
        <groupId>io.quarkus</groupId>
        <artifactId>quarkus-universe-bom</artifactId>
        <version>${quarkus.platform.version}</version>
        <type>pom</type>
        <scope>import</scope>
      </dependency>
...
    </dependencies>
  </dependencyManagement>
```

This bom also contains a dependency version for nimbus-jose-jwt, the most recent one (2.2.2.Final) contains the version 9.10 which is the correct version for strimzi-kafka-oauth.

The older versions, however, contain an older version of nimbus-jose-jwt (version 4.41.1 for example).

Unless the dependency version is explicitly overridden the one from bom is used.
One simple fix is to just add an override like this:

```
  <dependencyManagement>
    <dependencies>
...
      <dependency>
        <groupId>com.nimbusds</groupId>
        <artifactId>nimbus-jose-jwt</artifactId>
        <version>9.10</version>
      </dependency>
...
    </dependencies>
  </dependencyManagement>
 ```

I created an [example of using the OAuth authentication with Kafka](https://github.com/mstruk/getting-started-quarkus-kafka/tree/oauth) that includes native build.

**Comment by zbendhiba** [comment]
Hello @mstruk  yes, there was a problem coming from the camel-quarkus bom. I fixed it for  quarkus 2.1.x. 
You should upgrade the example to the latest version of quarkus.
